File: C#_LeftToDo/src/Tasks/SimpleTask.cs
using System;

namespace LeftToDo
{
    public class SimpleTask : Task // Derives from Task
    {
        public SimpleTask(string specification)
        {
            type = "S";
            description = specification;
        }
        // Displaying simple Task
        internal static void ShowTask(Task task, int index)
        {
            if (!task.done)
            {
                Console.WriteLine($"[ ]\t{index}\t{task.description}");
                return;
            }
            Console.WriteLine($"[X]\t{index}\t{task.description}");
        }
        // Overload, Displaying simple Task in Checklist
        internal static void ShowTask(Task task, int outer, int inner)
        {
            if (!task.done)
            {
                Console.WriteLine($"\t[ ]\t{outer} - {inner}\t{task.description}");
                return;
            }
            Console.WriteLine($"\t[X]\t{outer} - {inner}\t{task.description}");
        }
        // Takes info from user
        internal override void Create()
        {
            Console.WriteLine("Ange uppgift:");
            description = Console.ReadLine();
        }
    }
}

File: Go_leftToDo/tasks/simpleTask.go
package tasks

import (
	"fmt"
)

type SimpleTask struct {
	Task
	MarkAsDoneFn func(simple SimpleTask) SimpleTask
}

func NewSimpleTask(description string) *SimpleTask {
	s := SimpleTask{}
	s.TaskType = "S"
	s.Done = false
	s.Description = description
	return &s
}

func (s SimpleTask) Create() *SimpleTask {
	fmt.Printf("\nAnge uppgift:\n")
	var input string
	fmt.Scanln(&input)
	simple := NewSimpleTask(input)
	return simple
}

func (s *SimpleTask) ShowSubTask(outer int, inner int) {
	if !s.Done {
		fmt.Println("[ ]\t", outer, " - ", inner, "\t", s.Description)
	} else if s.Done {
		fmt.Println("[X]\t", outer, " - ", inner, "\t", s.Description)
	}
}

func (s *SimpleTask) MarkAsDone(simple SimpleTask) SimpleTask {
	return simple.MarkAsDoneFn(simple)
}

func (s SimpleTask) CreateMarkAsDone(simple SimpleTask) SimpleTask {
	simple.MarkAsDoneFn = func(simple SimpleTask) SimpleTask {
		return simple
	}
	return simple
}
